module/VuFind/src/VuFind/AjaxHandler/GetItemStatuses.php

```diff
@@ -219,17 +214,47 @@ protected function reduceServices(array $rawServices)
     }
 
     /**
-     * Create a delimited version of the call number to allow the Javascript code
-     * to handle the prefix appropriately.
+     * Create an array with the callnumber and prefix of the given item.
      *
-     * @param string $prefix     Callnumber prefix or empty string.
-     * @param string $callnumber Main call number.
+     * @param array $item Item's holding data.
+     *
+     * @return array      Associative array with the keys 'prefix' and 'callnumber'
+     */
+    protected function getCallNumberArray(array $item): array
+    {
+        return [
+            'prefix' => $item['callnumber_prefix'] ?? '',
+            'callnumber' => $item['callnumber'],
+        ];
+    }
+
+    /**
+     * Render the callnumber HTML.
+     *
+     * @param string $callnumberSetting The callnumber mode setting
+     * @param array  $callnumbers       Callnumbers to render
      *
      * @return string
      */
-    protected function formatCallNo($prefix, $callnumber)
+    protected function renderCallnumbers(string $callnumberSetting, array $callnumbers): string
     {
-        return !empty($prefix) ? $prefix . '::::' . $callnumber : $callnumber;
+        $html = [];
+
+        $callnumberHandler = $this->getCallnumberHandler($callnumbers, $callnumberSetting);
+        foreach ($callnumbers as $number) {
+            $displayCallnumber = $actualCallnumber = $number['callnumber'];
+
+            if (!empty($number['prefix'])) {
+                $displayCallnumber = $number['prefix'] . ' ' . $displayCallnumber;
+            }
+
+            $html[] = $this->renderer->render(
+                'ajax/itemCallnumber',
+                compact($actualCallnumber, $displayCallnumber, $callnumberHandler)
```

Comment:
Syntax is wrong (though an easy mistake to make, since this is a weird function); should be:

```suggestion
                compact('actualCallnumber', 'displayCallnumber', 'callnumberHandler')
```

Reply:
I fixed this.

On another note: I just created a new branch from upstream/dev but it's already 98 commits ahead without doing anything. How can this be?

Reply:
This PR is now targeted against dev-11.0 rather than dev. I'm trying to merge dev into dev-11.0 frequently, but is it possible the anomaly your seeing is related to this?

Reply:
My error. I should have synced the dev branch in my fork...

Reply:
That was about a new branch i created for another possible PR. The problem was that the dev branch in my fork wasn't up-to-date and i created the branch from upstream. It's alright now.

If I check out the upstream's dev branch, I get an error after logging in - `Laminas\ServiceManager\Exception\ServiceNotFoundException : Unable to resolve service "VuFind\Role\PermissionProvider\PluginManager" to a factory; are you certain you provided it during configuration?`
It seems the error exists since merging #3882 or is it just me?

Reply:
Does clearing out your $VUFIND_LOCAL_DIR/cache directory make a difference? Sometimes old cached configurations can cause this kind of problem. If that doesn't help, let me know and I'll see if I can get this branch running in my own environment.

Reply:
Thanks! Now it works!